### DISCUSSION TITLE
Upload from Modal not working

### DISCUSSION THREAD

**Discussion by wsdnathan** [orchidsoftware/platform#1951] [Q&A]
When I try to upload a file from a modal, I cannot access it because `$request->file('upload')` returns null.

Layout modal:
`            Layout::modal('importModal', Layout::rows([
                Upload::make('upload')
                    ->title('Upload CSV')
                    ->maxFiles(1)
                    ->acceptedFiles('.csv')
                    ->required(),
            ]))->title('Import CSV'),`

Import method:
`    public function importCSV(Request $request) {
        $file = new File($request->file('upload'));
        $attachment = $file->load();`

**Error:
`TypeError
Orchid\Attachment\File::__construct(): Argument #1 ($file) must be of type Illuminate\Http\UploadedFile, null given`**

**Comment by gladiiss** [comment]
Try this

```
public function importCSV(Request $request)
    {
        if (!$request->get('upload')) {
            Toast::error("Nothing uploaded!");
        }else{
            // get first attachment
            $attachment = Attachment::find($request->get('upload')[0]);
            // do whatever you want with attachment
        }
    }
```
